File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/purchasing/purchase-order-detail/purchase-order-detail-lookup-model.ts
import { IProductLookupModel } from 'app/models/data/entities/production/product/product-lookup-model';
import { IPurchaseOrderHeaderLookupModel } from 'app/models/data/entities/purchasing/purchase-order-header/purchase-order-header-lookup-model';

export interface IPurchaseOrderDetailLookupModel {
  purchaseOrderID: number;
  purchaseOrderDetailID: number;
  dueDate: Date | string;
  orderQty: number;
  productID: number;
  unitPrice: number;
  lineTotal: number;
  receivedQty: number;
  rejectedQty: number;
  stockedQty: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class PurchaseOrderDetailLookupModel implements IPurchaseOrderDetailLookupModel {
  purchaseOrderID: number;
  purchaseOrderDetailID: number;
  dueDate: Date | string;
  orderQty: number;
  productID: number;
  unitPrice: number;
  lineTotal: number;
  receivedQty: number;
  rejectedQty: number;
  stockedQty: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<PurchaseOrderDetailLookupModel>) {
    if (!!init) {
      Object.assign<PurchaseOrderDetailLookupModel, Partial<PurchaseOrderDetailLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Purchasing/PurchaseOrderDetail/Commands/Queries/GetPurchaseOrderDetails/PurchaseOrderDetailLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Purchasing;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Production.Product.Queries.GetProducts;
namespace AdventureWorks.Application.DataEngine.Purchasing.PurchaseOrderDetail.Queries.GetPurchaseOrderDetails
{
    public partial class PurchaseOrderDetailLookupModel : IHaveCustomMapping
    {
        public int PurchaseOrderID { get; set; }
        public int PurchaseOrderDetailID { get; set; }
        public DateTime DueDate { get; set; }
        public short OrderQty { get; set; }
        public int ProductID { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal LineTotal { get; set; }
        public decimal ReceivedQty { get; set; }
        public decimal RejectedQty { get; set; }
        public decimal StockedQty { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Purchasing.PurchaseOrderDetail, PurchaseOrderDetailLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<PurchaseOrderDetailLookupModel, BasePurchaseOrderDetail>().IgnoreMissingDestinationMembers();
        }
    }
}
